shell/server/core.js

```diff
@@ -140,11 +141,13 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
+    this.saved = !!saveTemplate.parentToken;  // TODO(cleanup): ugly
```

Comment:
I'm confused about what's going on here. Could you add a comment about what the `saved` field actually means and what `parentToken` has to do with it?